storybook(fxa-settings): Recreate AccountRecoveryConfirmKey in React

vpomerleau opened the pull request:
## Because

- We want to convert content-server views to React and are starting by building out the front-end in Storybook. This PR converts the account_recovery_confirm_key page. Additional functionality  will be added in later tickets.

## This pull request

- Create a new AccountRecoveryConfirmKey page with storybook, mocks, tests.

## Issue that this pull request solves

Closes: #FXA-6342

## Checklist

_Put an `x` in the boxes that apply_

- [ ] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

Previous (content-server)
![image](https://user-images.githubusercontent.com/22231637/210664368-34cb531f-a1c5-456d-b1ef-cce718253635.png)

(New) Storybook
![image](https://user-images.githubusercontent.com/22231637/210664342-5291a956-80f7-4870-b2f5-95b3b652fe53.png)

## Other information (Optional)

Any other information that is important to this pull request.

bcolsson commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/en.ftl:
```diff
@@ -0,0 +1,22 @@
+# { $serviceName } represents a product name (e.g., Mozilla VPN) that will be passed in as a variable
+account-recovery-confirm-key-heading-w-custom-service = Reset password with account recovery key <span>to continue to { $serviceName }</span>
+
+account-recovery-confirm-key-instructions = Please enter the one time use account recovery key you stored in a safe place to regain access to your Firefox Account.
```
```suggestion
account-recovery-confirm-key-instructions = Please enter the one time use account recovery key you stored in a safe place to regain access to your { -product-firefox-account }.
```
Use the term here. (Note, account should no longer be capitalized.)

bcolsson commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/en.ftl:
```diff
@@ -0,0 +1,22 @@
+## AccountRecoveryConfirmKey page
+
+# Strings within the <span> elements appear as a subheading.
+# If more appropriate in a locale, the string within the <span>, "to continue to { $serviceName }" can stand alone as "Continue to { $serviceName }"
```
```suggestion
# If more appropriate in a locale, the string within the <span>, "to continue to account settings" can stand alone as "Continue to account settings"
```

millsoper replied:
If the service name is variable, should we still have "account settings" in the comment?

millsoper replied:
Ohhhh but it's hard-coded in below, I understand now!

bcolsson commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/en.ftl:
```diff
@@ -0,0 +1,22 @@
+# Clicking this button checks if the recovery key provided by the user is correct and associated with their account
+account-recovery-confirm-key-button = Confirm account recovery key
+# Error displayed in an alert banner when the recovery key confirmation is unsuccessful
+account-recovery-confirm-key-error-general = Recovery code is not valid
```
Should this be `Account recovery key` instead of `Recovery code`?

vpomerleau replied:
Yes, you're absolutely right!

@millsoper The original error message was `invalid hex string: null` which seems seems as a user-facing error message (in alert bar) - do you think `Account recovery key is not valid` is a reasonable alternative? Maybe this need product review?

millsoper replied:
That's a great question -- I think it seems like a valid alternative but I would love a product review. I'll ping Vesta!

bcolsson replied:
I approved the PR as this string doesn't necessarily block localization (but would still hold off merging until you hear back from Vesta).

vpomerleau replied:
@millsoper since this route won't be active for a while yet (and will be reviewed again when GQL is stitched in), do you think we can tentatively keep this updated error message with the comment in `index.tsx`?

millsoper commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/en.ftl:
```diff
@@ -0,0 +1,22 @@
+## AccountRecoveryConfirmKey page
```
Oh good to know, I didn't know this!

vpomerleau replied:
I don't think any of our en.ftl files have an empty line at the top since it gets added in by default by the grunt concat task. Might not be necessary unless we're adding a group comment right after another line?

bcolsson replied:
@vpomerleau @millsoper 
Looks like that is in fact the case. Please ignore this comment then!

vpomerleau replied:
Still good to know if we ever add a group comment within the file. Thanks @bcolsson!

millsoper commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/index.tsx:
```diff
@@ -0,0 +1,169 @@
+import WarningMessage from '../../components/WarningMessage';
+import ResetPasswordLinkExpired from '../../components/ResetPasswordLinkExpired';
+
+// --serviceName-- is the relying party
```
👍 nice

millsoper commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/index.tsx:
```diff
@@ -0,0 +1,169 @@
+  recoveryKey: string;
+};
+
+type LinkStatus = 'expired' | 'valid';
```
Does this not have a damaged link status?

vpomerleau replied:
It doesn't look like the original content-server view for this component included a "damaged link" case. Could be good to add though?

millsoper replied:
That's so curious...I guess it might be good to add it in?

millsoper commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/index.stories.tsx:
```diff
@@ -0,0 +1,37 @@
+export default {
+  title: 'pages/AccountRecoveryConfirmKey',
+  component: AccountRecoveryConfirmKey,
+} as Meta;
```
It really makes me laugh that we call this `Meta` for a facebook product

vpomerleau replied:
That's so meta!

millsoper commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/index.tsx:
```diff
@@ -0,0 +1,169 @@
+    } catch (e) {
+      const errorAccountRecoveryConfirmKey = ftlMsgResolver.getMsg(
+        'account-recovery-confirm-key-error-general',
+        // Original error message was 'invalid hex string: null'
```
We can probably remove this comment after getting product confirmation

millsoper commented on packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/index.tsx:
```diff
@@ -0,0 +1,169 @@
+
+  return (
+    <>
+      {linkStatus === 'expired' && <ResetPasswordLinkExpired />}
```
noice 🚀

millsoper reviewed (approved):
This looks great! I think we should wait for product approval for the new error message if possible, and we can consider adding in a damaged link view, but otherwise good to go!

vpomerleau commented:
FYI @bcolsson, some additional changes to l10n:

I spotted missing l10n for an error tooltip, added it just now as `account-recovery-confirm-key-empty-input-error`.

Also, updated the comment for `reset-password-heading-w-default-service`

vpomerleau commented:
> This looks great! I think we should wait for product approval for the new error message if possible, and we can consider adding in a damaged link view, but otherwise good to go!

Awesome, thanks for the review!! I've addressed the changes and added in a `damaged` state.

I think we're just pending approval for the error messaging.
